packages/components/src/FormatValue/FormatValue.component.js

```diff
@@ -128,48 +127,48 @@ function replaceWhiteCharacters(content, regexp, t) {
 	return splitting.filter(isEmptyCharacter).map(value => replaceCharacterByIcon(value, t));
 }
 
-export function FormatValueComponent(props) {
-	let content = [];
+export function FormatValueComponent({ value, className }) {
+	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
 
-	const hiddenCharsRegExpMatch = props.value.match(REG_EXP_LEADING_TRAILING_WHITE_SPACE_CHARACTERS);
+	let formattedValue = value;
+	if (hasWhiteSpaceCharacters(value)) {
+		let content = [];
+		const hiddenCharsRegExpMatch = value.match(REG_EXP_LEADING_TRAILING_WHITE_SPACE_CHARACTERS);
 
-	if (hiddenCharsRegExpMatch[1]) {
-		content = replaceWhiteCharacters(
-			hiddenCharsRegExpMatch[1],
-			REG_EXP_REPLACED_WHITE_SPACE_CHARACTERS,
-			props.t,
-		);
-	}
+		if (hiddenCharsRegExpMatch[1]) {
+			content = replaceWhiteCharacters(
+				hiddenCharsRegExpMatch[1],
+				REG_EXP_REPLACED_WHITE_SPACE_CHARACTERS,
+				t,
+			);
+		}
 
-	if (hiddenCharsRegExpMatch[2]) {
-		const formattedContent = replaceWhiteCharacters(
-			hiddenCharsRegExpMatch[2],
-			REG_EXP_CAPTUR_LINE_FEEDING,
-			props.t,
-		);
-		content = [...content, ...formattedContent];
-	}
+		if (hiddenCharsRegExpMatch[2]) {
+			const formattedContent = replaceWhiteCharacters(
+				hiddenCharsRegExpMatch[2],
+				REG_EXP_CAPTUR_LINE_FEEDING,
+				t,
+			);
+			content = [...content, ...formattedContent];
+		}
+
+		if (hiddenCharsRegExpMatch[3]) {
+			const formattedContent = replaceWhiteCharacters(
+				hiddenCharsRegExpMatch[3],
+				REG_EXP_REPLACED_WHITE_SPACE_CHARACTERS,
+				t,
+			);
+			content = [...content, ...formattedContent];
+		}
 
-	if (hiddenCharsRegExpMatch[3]) {
-		const formattedContent = replaceWhiteCharacters(
-			hiddenCharsRegExpMatch[3],
-			REG_EXP_REPLACED_WHITE_SPACE_CHARACTERS,
-			props.t,
-		);
-		content = [...content, ...formattedContent];
+		formattedValue = <span className={className}>{content.map(addKeyAttribute)}</span>;
 	}
-
-	return <span className={props.className}>{content.map(addKeyAttribute)}</span>;
+	return formattedValue;
 }
 
 FormatValueComponent.propTypes = {
 	className: PropTypes.string,
 	value: PropTypes.string,
-	t: PropTypes.func,
-};
-
-FormatValueComponent.defaultProps = {
-	t: getDefaultT(),
 };
 
-export default withTranslation(I18N_DOMAIN_COMPONENTS)(FormatValueComponent);
```

Comment:
if you do not wrap how context is provided ?

Reply:
I replaced it with the hook, like half of react-components components